### PR TITLE
Feature/replace-backup-file-by-seed-phrase

### PR DESCRIPTION
**Pull request by janek26**
https://user-images.githubusercontent.com/6831124/162443094-b0ff288d-5821-4b85-97af-dee23b1969b6.mp4

### CODE REVIEW

File: packages/extension/src/background/wallet.ts [hunk 1]
```diff
@@ -64,6 +64,18 @@ export class Wallet extends EventEmitter {
     return this.session !== undefined
   }
 
+  public async getSeedPhrase(): Promise<string> {
+    if (!this.isSessionOpen() || !this.session || !this.encryptedBackup) {
```
Comment (delaaxe) [inline packages/extension/src/background/wallet.ts:68]: Looks like we're not so confident on the invariants this class is supposed to hold

File: packages/extension/src/ui/App.tsx [hunk 2]
```diff
@@ -83,14 +85,10 @@ const Screen: FC = () => {
...
   const { isLoading } = useAppState()
   const { actions } = useActions()
-  const { isBackupDownloadRequired } = useBackupDownload()
 
   if (isLoading) {
     return <LoadingScreen />
   }
-  if (isBackupDownloadRequired) {
-    return <BackupDownloadScreen />
-  }
```
Comment (delaaxe) [inline packages/extension/src/ui/App.tsx:93]: so no more backup file at all?

File: packages/extension/src/ui/App.tsx [hunk 3]
```diff
@@ -114,6 +112,18 @@ const Screen: FC = () => {
...
+            element={<ConfirmSeedRecoveryPage />}
+          />
+          <Route
+            path={routes.setupSeedRecovery()}
+            element={<SetupSeedRecoveryPage />}
+          />
+          <Route
+            path={routes.setupRecovery()}
+            element={<SetupRecoveryPage />}
+          />
```
Comment (delaaxe) [inline packages/extension/src/ui/App.tsx:126]: we should increase max line width

File: packages/extension/src/ui/components/Account/UpdateBanner.tsx [hunk 4]
```diff
@@ -1,70 +1,38 @@
...
-        <UpdateBannerDescription>
-          This may have breaking changes
-        </UpdateBannerDescription>
-      </UpdateTextWrapper>
-    </UpdateBannerWrapper>
+    <Banner
+      title="Update Available"
+      description="This may have breaking changes"
+      icon={
+        <svg
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Account/UpdateBanner.tsx:15]: Should be in its own file

File: packages/extension/src/ui/components/Banner.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  cursor: pointer;
+  align-items: center;
+  padding: 16px;
+  background-color: #ffffff;
+  border-radius: 8px;
+  ${({ noMargins = false }) =>
+    !noMargins &&
+    css`
+      margin: 16px 20px;
+    `};
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Banner.tsx:17]: `styled-components` is really an awful css library

File: packages/extension/src/ui/components/FeeEstimate.tsx [hunk 6]
```diff
@@ -3,7 +3,6 @@ import DangerIcon from "@mui/icons-material/ReportGmailerrorredRounded"
 import Tippy from "@tippyjs/react"
```
Comment (delaaxe) [inline packages/extension/src/ui/components/FeeEstimate.tsx:3]: component and filename should either be called `FeeEstimation` or `EstimateFee`

File: packages/extension/src/ui/components/Recovery/IconBar.tsx [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  align-items: center;
+  justify-content: space-between;
+  padding: 18px;
+`
+
+const CloseIcon: FC<{
+  onClick?: () => void
+}> = ({ onClick }) => {
+  return (
+    <svg
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Recovery/IconBar.tsx:18]: I'd rather put all svgs in their own files

File: packages/extension/src/ui/components/Recovery/IconBar.tsx [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+        />
+      ) : (
+        <div />
+      )}
+      <CloseIcon
+        onClick={() => {
+          navigate(routes.account())
+        }}
+      />
+    </IconBar>
```
Comment (delaaxe) [inline packages/extension/src/ui/components/Recovery/IconBar.tsx:90]: spaghetti

File: packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+      &:hover {
+        background-color: rgba(255, 255, 255, 0.4);
+      }
+    `}
+`
+
+function getRandomWord(excludeWordlist: string[]): string {
+  const excludeIndexes = excludeWordlist.map((word) =>
+    wordlists.en.getWordIndex(word),
+  )
```
Comment (delaaxe) [inline packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx:78]: be more functional when you can, have you tried:

```suggestion
  const excludeIndexes = excludeWordlist.map(wordlists.en.getWordIndex)
```

File: packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx [hunk 10]
```diff
@@ -0,0 +1,233 @@
...
+      ? selectedWords.filter((selectedWord) => selectedWord !== word)
+      : [...selectedWords, word]
+
+    setSelectedWords(updatedSelectedWords)
+
+    const expectedWords = wordlists.en
+      .split(seedPhrase)
+      .filter((word) => confirmSeedPhrase.includes(word))
+
+    console.log(expectedWords, updatedSelectedWords)
```
Comment (delaaxe) [inline packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx:146]: console.log

File: packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx [hunk 11]
```diff
@@ -0,0 +1,233 @@
...
+              }
+            }}
+            disabled={showError}
+          >
+            Confirm
+          </Button>
+          {showError && (
+            <FormError
+              style={{
+                marginTop: "8px",
```
Comment (delaaxe) [inline packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx:222]:
```suggestion
                marginTop: 8,
```

File: packages/extension/src/ui/screens/SetupSeedRecovery.tsx [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+    () => getSeedPhrase(),
+    {
+      suspense: true,
+    },
+  )
+
+  return (
+    <SeedPhraseGrid>
+      {seedPhrase.split(" ").map((word, index) => (
+        <SeedWordBadge key={index}>{word}</SeedWordBadge>
```
Comment (delaaxe) [inline packages/extension/src/ui/screens/SetupSeedRecovery.tsx:64]: it's generally a bad idea to use `index` as react key, why can't we use word as index here?

File: packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+      suspense: true,
+    },
+  )
+
+  const { data: confirmSeedPhrase = [] } = useSWRImmutable(
+    "confirmSeedPhrase",
+    () => createConfirmationWordlist(wordlists.en.split(seedPhrase)),
+    {
+      suspense: true,
+    },
```
Comment (delaaxe) [inline packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx:131]:
```suggestion
    { suspense: true },
```

File: packages/extension/src/background/wallet.ts (hunk 1 above)
Comment (janek26) [reply packages/extension/src/background/wallet.ts:68]: this.isSessionOpen is the only relevant check here. If that's true, both of the following are true aswell.
It was just to trick the TS compiler. If there's a better way to do this let me know (`!` is forbidden by eslint)

File: packages/extension/src/ui/App.tsx (hunk 3 above)
Comment (janek26) [reply packages/extension/src/ui/App.tsx:126]: why? :D that would be a huge change to the entire codebase. I think both formats are readable

File: packages/extension/src/ui/components/Account/UpdateBanner.tsx (hunk 4 above)
Comment (janek26) [reply packages/extension/src/ui/components/Account/UpdateBanner.tsx:15]: agree :)

File: packages/extension/src/ui/components/Recovery/IconBar.tsx (hunk 8 above)
Comment (janek26) [reply packages/extension/src/ui/components/Recovery/IconBar.tsx:90]: solved, thanks :)

File: packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx (hunk 10 above)
Comment (janek26) [reply packages/extension/src/ui/screens/ConfirmSeedRecovery.tsx:146]: thanks

File: packages/extension/src/ui/App.tsx (hunk 2 above)
Comment (janek26) [reply packages/extension/src/ui/App.tsx:93]: We'll keep it for now in the settings for people to download :) We'll also keep the recovery by backup file for a few more versions

Comment (janek26) [reply packages/extension/src/ui/App.tsx:93]: you can find it a little further down: 
```
<Route
  path={routes.backupDownload()}
  element={<BackupDownloadScreen />}
/>
```

File: packages/extension/src/ui/screens/SetupSeedRecovery.tsx (hunk 12 above)
Comment (janek26) [reply packages/extension/src/ui/screens/SetupSeedRecovery.tsx]: we need both as an id now, as the number (index) will display in the future

**Comment by janek26** [comment]
waiting for #524

**Review by gergold — APPROVED** [review]

**Pull request merged**